Backend/auction-app/src/main/java/com/example/auctionapp/model/User.java

```diff
@@ -0,0 +1,96 @@
+package com.example.auctionapp.model;
+
+
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import java.util.List;
+
+@Entity
+@Table(name = "user_entity")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank(message = "First name can't be blank")
+    @Column(name = "first_name")
+    private String firstName;
+
+    @NotBlank(message = "Last name can't be blank")
+    @Column(name = "last_name")
+    private String lastName;
+
+    @Email(message = "Email must be valid")
+    private String email;
+
+    @NotBlank
+    private String password;
+
+    @ManyToMany
+    private List<Role> roles;
+
+    public User() {
+    }
+
+    public User(Long id, @NotBlank(message = "First name can't be blank") String firstName,
+                @NotBlank(message = "Last name can't be blank") String lastName,
+                @Email(message = "Email must be valid") String email,
+                @NotBlank String password, List<Role> roles) {
```

Comment:
You already used these annotations (@NotBlank, @Email etc) on the properties, no need to put them in the constructor arguments as well.
Also, code formatting, each method argument should go into a new line

Reply:
I dont see password `hash` and `salt` fields anywhere.